# Predicting Political Affiliation

In this notebook I attempt to build a model to predict a persons' party afilliation.  For simplicity's sake, we will exclude the minor parties and only attempt to predict whether or not a person is a Republican or Democrat.  Since we're predicting a binary categorical variable, we can use either Logisitic Regression or Linear Discriminant Analysis to achieve our end goal. 

---

## Data

The data used in this analysis comes from [Harvard/YouGov](https://cces.gov.harvard.edu/).  The data is a representative sample of 18,200 American adults taken in 2017.

In [18]:
import pandas as pd
import seaborn as sns

path = "data/Common Content Data.tab"
path1 = "data/CCES17_Common_county.tab"

data = pd.read_csv(path, sep="\t", nrows=1000)
location_data = pd.read_csv(path1, sep="\t", nrows=1000)

data = pd.merge(data, location_data, on='V101')

data.head()

V101  weights_common  cdid115  lookupzip      CurrentGovName  \
0  361018952        1.055111        5      85298          Doug Ducey   
1  361182426        0.335574        2       1002    Charles D. Baker   
2  360873135        0.930851        1      19802  John C. Carney Jr.   
3  361039382        0.931859        4      11566        Andrew Cuomo   
4  360999746        0.437988        1       1075    Charles D. Baker   

  CurrentGovParty      CurrentHouseName CurrentHouseParty CurrentSen1Name  \
0      Republican            Andy Biggs        Republican      Jeff Flake   
1      Republican          Jim McGovern        Democratic       Ed Markey   
2      Democratic  Lisa Blunt Rochester        Democratic     Chris Coons   
3      Democratic         Kathleen Rice        Democratic   Chuck Schumer   
4      Republican          Richard Neal        Democratic       Ed Markey   

  CurrentSen1Party      ...       healthins_5 healthins_6 healthins2 phone  \
0       Republican      ...                 2           2          9     1   
1       Democratic      ...                 2           2          9     3   
2       Democratic      ...                 2           2          9     3   
3       Democratic      ...                 2           2          9     3   
4       Democratic      ...                 2           2          9     2   

  internethome  internetwork                starttime  \
0            1             3  2017-11-14 06:50:48.000   
1            1             3  2017-11-15 16:17:41.000   
2            1             3  2017-11-14 06:50:01.000   
3            1             3  2017-11-14 14:29:06.000   
4            1             1  2017-11-14 15:51:03.000   

                   endtime  countyfips     countyname  
0  2017-11-14 07:40:12.000      4013.0    Maricopa AZ  
1  2017-11-18 01:58:14.000     25015.0   Hampshire MA  
2  2017-11-14 07:47:43.000     10003.0  New Castle DE  
3  2017-11-14 15:11:51.000     36059.0      Nassau NY  
4  2017-12-03 05:40:05.000     25015.0   Hampshire MA  

[5 rows x 269 columns]

The first thing to do is to narrow down excatly which variables which would be useful for analaysis.  The dataset currently contains 269 columns so  first we need to review the data documentation and select the variables that would be interesting when conducting our analysis.  After that we can perform some dimmension reduction technique to further narrow down our predictor variables if needed.

In [24]:
'inputzip' in data.columns

True